File: src/main/java/me/fourteendoggo/xkingdoms/storage/persistence/PersistenceHandler.java
package me.fourteendoggo.xkingdoms.storage.persistence;

import me.fourteendoggo.xkingdoms.player.KingdomPlayer;
import me.fourteendoggo.xkingdoms.utils.Home;

import java.util.UUID;

public interface PersistenceHandler {

    KingdomPlayer loadPlayer(UUID id);

    void deleteHome(Home home);

    void savePlayer(KingdomPlayer player);
}

File: src/main/java/me/fourteendoggo/xkingdoms/storage/persistence/PersistenceHandler.kt
package me.fourteendoggo.xkingdoms.storage.persistence

import me.fourteendoggo.xkingdoms.utils.Home
import java.util.*

interface PersistenceHandler {
    fun loadPlayer(id: UUID): KingdomPlayer?
    fun deleteHome(home: Home)
    fun savePlayer(player: KingdomPlayer)
}
